src/tests.zig

```diff
@@ -156,6 +156,8 @@ test "serializes a structure with variable fields" {
     defer list.deinit();
     try serialize(@TypeOf(data), data, &list);
     try expect(std.mem.eql(u8, list.items, serialized_data[0..]));
+    var out: @TypeOf(data) = undefined;
```

Comment:
The issue with this PR is that `.name` and `.company` are `comptime` and you can't take a runtime reference to them. This is because they are understood as `[5]u8` and `[8]u8` and not `[]const u8`. Casting with `@as` solves the problem, but this is going to cause problems. More digging needs to be done.

Reply:
This comes from the interpretation of a dereference,